### ISSUE TITLE
Build | Document should use roclets defined in DESCRIPTION if available

### ISSUE THREAD

**Issue by hadley** [rstudio/rstudio#6433]
### System details

    RStudio Edition : Desktop [Open Source]
    RStudio Version : 1.3.897
    OS Version      : macOS  10.15.3
    R Version       : R version 3.6.0 (2019-04-26)

### Describe the problem in detail

Packages now define which roclets should be run by using the `roclets` element of the `Roxygen` `DESCRIPTION` field. But RStudio overrides this to use a field in the project specific metadata. It would be nice to be able to opt out of RStudio managing this field.

**Comment by mstr3336** [comment]
Seconding this and presenting a workaround: 

`devtools` also implements a documentation addin:

| Name | Shortcut | Scope |
| - | - | - |
| Document a package |  | Addin |

This calls `devtools::document()` which wraps `roxygen2::roxygenize()`, but the `devtools` addin does not override parameters in this call, so the `Roxygen` `DESCRIPTION` field, and the `man/roxygen/meta.R` settings are respected.

**Comment by matthewstrasiotto** [comment]
please don't close this

**Comment by kevinushey** [comment]
@hadley is this still worth looking at?

**Comment by romainfrancois** [comment]
We can look at the result of `roxygen2:::load_options_description()` and check for `$roclets` there.

**Comment by hadley** [comment]
Yeah, this probably worth fixing. @romainfrancois if you tackle this, can you file an issue in roxygen2 so that we make an exported function for this? (Just so we don't accidentally break in roxygen2 at some point in the future)

**Comment by hadley** [comment]
OTOH it might be better to wait until after https://github.com/r-lib/roxygen2/issues/1328, when you'll be able to parse the DESCRIPTION yourself:

```yaml
Config/roxygen2/roclets: rd, namespace
```

**Comment by romainfrancois** [comment]
Would this still need to take care of this syntax so that it would work for older roxygen2 ?

```
Roxygen: list(markdown = TRUE, roclets = c("rd"))
```

`roxygen2:::load_options_description()` is mostly `read.dcf()`, we could support both versions: 

```r
> roxygen2:::load_options_description
function (base_path = ".") 
{
    desc_path <- file.path(base_path, "DESCRIPTION")
    dcf <- read.dcf(desc_path, fields = c("Roxygen", "Package"))
    desc_opts <- dcf[[1, 1]]
    if (is.na(desc_opts)) {
        opts <- list()
    }
    else {
        opts <- eval(parse(text = desc_opts), child_env(baseenv()))
    }
    opts$current_package <- dcf[[1, 2]]
    opts
}
<bytecode: 0x120a8b040>
<environment: namespace:roxygen2>
```

**Comment by hadley** [comment]
If we change syntax, roxygen2 would probably update on first run, so I think you wouldn't need to worry about supporting older versions in the IDE.

**Comment by t-kalinowski** [comment]
I just ran into this and was about to open an issue if one wasn't already open. 

I think the fix is relatively simple, mostly deleting code. Right now, RStudio does extra work to populate the arguments to the `devtools::document(...)` call. 

Instead, RStudio can simply call `devtools::document()` with no arguments, which already does the right thing.
